packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+    })
+
+    if (restart) {
+      isRestarted = true
+      const { status }: FeelessVerificationState = yield select(feelessVerificationStateSelector)
+      ValoraAnalytics.track(VerificationEvents.verification_resend_messages, {
+        count: status.numAttestationsRemaining,
+      })
+    } else {
+      return verification
+    }
+    Logger.debug(TAG, 'Verification has been restarted')
+  }
+}
+
+export function* feelessDoVerificationFlow(withoutRevealing: boolean = false) {
+  Logger.debug(TAG, '@feelessDoVerificationFlow', 'Starting feeless verification')
+  let receiveMessageTask: Task | undefined
+  let autoRetrievalTask: Task | undefined
+
+  try {
+    const [walletAddress, contractKit, e164Number]: [string, ContractKit, string] = yield all([
+      call(getConnectedUnlockedAccount),
+      call(getContractKit),
+      select(e164NumberSelector),
+      put(feelessSetVerificationStatus(VerificationStatus.Prepping)),
+    ])
+
+    let feelessVerificationState: FeelessVerificationState = yield select(
+      feelessVerificationStateSelector
+    )
+
+    const komenciKit = new KomenciKit(contractKit, walletAddress, {
+      url: KOMENCI_URL,
+      token: feelessVerificationState.komenci.sessionToken,
+    })
+
+    // Start by checking again to make sure Komenci is ready. Throws error if not
+    yield call(fetchKomenciReadiness, komenciKit)
+
+    // There should be no instances where the first param is truethy but the second isn't.
+    // Mainly including the second param to satisfy typescript
+    if (
+      !feelessVerificationState.status.isVerified ||
+      !feelessVerificationState.komenci.unverifiedMtwAddress
+    ) {
+      // Start or resume a Komenci session and update state. Throws error if unable to do so
+      yield call(startOrResumeKomenciSession, komenciKit, e164Number)
+
+      // Adds phoneHash into verification state, will ping Komenci for it only if needed.
+      // Throws an error if unable to get phoneHash
+      yield call(fetchPhoneHashDetails, komenciKit, e164Number)
+
+      const startingPepperQuota = feelessVerificationState.komenci.pepperQuotaRemaining
+      // Now that we are guarnateed to have the phoneHash, check again to see if the
+      // user already has a verified MTW
+      yield call(reFetchVerifiedMtw, contractKit, walletAddress, e164Number, startingPepperQuota)
+
+      const mtwIsVerified = yield call(isMtwVerified)
+
+      if (mtwIsVerified) {
+        yield call(endFeelessVerification)
+        return true
+      }
+
+      yield call(fetchOrDeployMtw, contractKit, komenciKit, walletAddress, e164Number)
+
+      // Registering the DEK and wallet before beginning verification to guarantee that every
+      // verified MTW address (i.e., accountAddress) has an EOA (i.e., walletAddress)
+      // registered to it. If that's not guaranteed, users are at risk of sending to MTWs
+      yield call(feelessDekAndWalletRegistration, komenciKit, walletAddress)
```

Comment:
Are we blocking on this transaction succeeding? IMO there's no reason no to batch this with the attestation request. Would save us 5-10s if this call is blocking..

Reply:
We are blocking on this transaction succeeding. This is what I mentioned in that thread with @nategraf about how I am attempting to guarantee that every verified MTW has a DEK and `walletAddress` registered to it in the Accounts contract.

I agree that there is an optimization opportunity around batching this tx, as there are with other transactions later in the flow (e.g., approve + request tx for requests). I am biased to keeping it as is for now and tackle optimizations within the context of the larger remodeling we'd like to do of this flow.

Reply:
Batching the approve and request tx *should* be a requirement enforced by Komenci... I agree that batching the DEK/wallet registration is a future optimization

Reply:
If Komenci isn't batching them then we have a big problem, but I would be very surprised if it's not